### ISSUE TITLE
VideoStreamPlayer causes random crashes on a Windows export without "Export with Debug"

### ISSUE THREAD

**Issue by Freecreativename** [godotengine/godot#78788]
### Godot version

4.1.beta3, 4.0.3.stable

### System information

Godot v4.1.beta3 - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3070 Ti (NVIDIA; 31.0.15.3598) - Intel(R) Core(TM) i5-8600K CPU @ 3.60GHz (6 Threads)

### Issue description

The game crashes on a Windows export without "Export with Debug" when I use VideoStreamPlayer node as a background in my scene.

I have created 2 simple scenes with VideoStreamPlayer and a button in each to switch between them. When I run the game in the editor, it works fine, when I export it with debug it works fine. As soon as I untick "Export with Debug" the game crashes randomly, sometimes after a few button clicks(changing scenes), sometimes on the first click... The console gives no errors whatsoever...

Tried to do it in both: Godot 4.0.3 and 4.1 beta3, the result is the same. 

I have also recreated this in Godot 3.5.2, the exported game without "Export with Debug" works fine, without crashes at all.

I have spent 4-5 hours yesterday searching internet and trying to fix it, but I couldn't. Tried different ffmpeg builds to convert the video, tried different video framerates, with audio and without audio... Looks like it's a Godot 4 issue.

### Steps to reproduce

1 - create a scene, add VideoStreamPlayer node, assign your video to it.
2 - create another scene with a VideoStreamPlayer node and assign your video to it.
3 - create buttons on each scene to switch between them.
4 - export your project for windows without "Export with Debug"
5 - play the game, after switching the scenes a couple of time, the game will crash.

### Minimal reproduction project

[Crash_Test_3.zip](https://github.com/godotengine/godot/files/11893147/Crash_Test_3.zip)

**Comment by akien-mga** [comment]
Tested the MRP on Linux and I can't reproduce a crash with the Linux release export template or the Windows release export template (through Wine), so this may be Windows specific.

With the Linux template, I get these errors raised occasionally when switching scenes, which can relate to what would be causing a crash on Windows:
```
ERROR: Condition "slot >= slot_max" is true. Returning: nullptr
   at: get_instance (./core/object/object.h:986)
```

**Comment by Freecreativename** [comment]
I have just played the exported game on Windows 8.1 and didn't get any crashes, although I have switched scenes about 100 times... Then I closed the game with Alt+F4 and reopened it. I got a crash after 3 clicks... Weird...


> Tested the MRP on Linux and I can't reproduce a crash with the Linux release export template or the Windows release export template (through Wine), so this may be Windows specific.
> 
> With the Linux template, I get these errors raised occasionally when switching scenes, which can relate to what would be causing a crash on Windows:
> 
> ```
> ERROR: Condition "slot >= slot_max" is true. Returning: nullptr
>    at: get_instance (./core/object/object.h:986)
> ```

What is causing this error? Do you know how to fix it?

**Comment by akien-mga** [comment]
This error signals an internal engine bug, it's not something you're doing wrong and can fix, at most you can try to change how you change scenes to work it around.

https://github.com/godotengine/godot/blob/9a09ee5d0217c4d5e5668fccddfa477bf16af357/core/object/object.h#L982-L986

**Comment by akien-mga** [comment]
You can do something like this:

```gdscript
func _on_button_pressed():
	var new_scn = load("res://scene_2.tscn").instantiate()
	$/root.add_child(new_scn)
	self.queue_free()
```

And likewise for `main.tscn`.

This adds the new scene to the root, and then frees the old scene after a delay, which seems to prevent the error in my test, so it might fix the crash on Windows.

`change_scene_to_file` seems to be too eager freeing the old scene while it's still in use maybe.

**Comment by Freecreativename** [comment]
> You can do something like this:
> 
> ```gdscript
> func _on_button_pressed():
> 	var new_scn = load("res://scene_2.tscn").instantiate()
> 	$/root.add_child(new_scn)
> 	self.queue_free()
> ```
> 
> And likewise for `main.tscn`.
> 
> This adds the new scene to the root, and then frees the old scene after a delay, which seems to prevent the error in my test, so it might fix the crash on Windows.
> 
> `change_scene_to_file` seems to be too eager freeing the old scene while it's still in use maybe.

Thanks for your reply! Sadly your workaround didn't fix random crashes, checked it on Windows 10 and Windows 8.1.

**Comment by YuriSizov** [comment]
Reopening, because https://github.com/godotengine/godot/pull/78987 is getting reverted.

**Comment by akien-mga** [comment]
This might have been fixed by #78988. At least for me the errors reported in https://github.com/godotengine/godot/issues/78788#issuecomment-1611195492 are no longer reproduced in 4.2-rc1.

Could you test to confirm whether 4.2-rc1 (or any later version when you get to test) solves the crash you had on Windows?

**Comment by akien-mga** [comment]
Based on my tests I'll close this as fixed by #78988, but please let us know if you still reproduce that crash in 4.2-rc1 or later.
